zerver/tests/test_push_notifications.py

```diff
@@ -1145,12 +1175,40 @@ def test_modernize_apns_payload(self) -> None:
         self.assertEqual(
             modernize_apns_payload(
                 {'alert': 'Message from Hamlet',
-                 'message_ids': [3]}),
+                 'message_ids': [3],
+                 'badge': 0}),
             payload)
         self.assertEqual(
             modernize_apns_payload(payload),
             payload)
 
+    @mock.patch('zerver.lib.push_notifications.push_notifications_enabled', return_value = True)
+    def test_apns_badge_count(self, mock_push_notifications: mock.MagicMock) -> None:
+        user_profile = self.example_user('othello')
+        # Test APNs badge count for personal messages.
+        message_ids = [self.send_personal_message(self.sender,
+                                                  user_profile,
+                                                  'Content of message')
+                       for i in range(5)]
+        self.assertEqual(get_apns_badge_count(user_profile), 5)
+        # Mark the messages as read and test whether
+        # the count decreases correctly.
+        for i, message_id in enumerate(message_ids):
+            do_update_message_flags(user_profile, get_client("website"), 'add', 'read', [message_id])
+            self.assertEqual(get_apns_badge_count(user_profile), 4-i)
```

Comment:
A small tweak: the relationships between all these numeric constants aren't obvious on a quick scan of this code, without reading it more carefully. So this would be clearer if that were made explicit: pull out the number of messages as a local variable, and then say like `num_messages - i - 1` for the last assert (and `num_messages` for both the setup and the first assert.)

Similarly for the other half of this test below.